api/src/main/java/io/druid/data/input/impl/FileIteratingFirehose.java

```diff
@@ -44,13 +47,26 @@ public FileIteratingFirehose(
   {
     this.lineIterators = lineIterators;
     this.parser = parser;
+    final ParseSpec parseSpec = parser.getParseSpec();
+    if (parseSpec instanceof CSVParseSpec) {
```

Comment:
Hmm. Returning nulls looks to affect to the query result in Druid. If so, I think returning nulls and skipping some rows are different. For example, a query for counting all rows will return different results.